ably/realtime_conn.go

```diff
@@ -360,21 +362,37 @@ func (c *Connection) connectWith(arg connArgs) (result, error) {
 			ConnectionStateDisconnected,
 		)
 	}
-	query, err := c.params(arg.mode)
-	if err != nil {
-		return nil, err
-	}
-	u.RawQuery = query.Encode()
-	proto := c.opts.protocol()
-
-	if c.State() == ConnectionStateClosed { // RTN12d - if connection is closed by client, don't try to reconnect
-		return nopResult, nil
-	}
+	var conn conn
+	host := c.hosts.getPreferredHost()
+	for {
+		u, err := url.Parse(c.opts.realtimeURL(host))
```

Comment:
It's incorrect to pass a fallback host into `opts.realtimeURL`, since it will reference `opts.Port` and `opts.TLSPort` which should only be used when connecting to the primary endpoint, not the fallbacks.

Reply:
It will reference only standard `PORT/TLSPORT`. We already have checks in place to make sure not to return fallbacks for custom port/TLSport set
https://github.com/ably/ably-go/blob/1405d1fc946be7f8649471f1f7059b4d92a5a30f/ably/options.go#L346-L369

Reply:
Ok, again, this really isn't very clear, please add a comment to this effect.

Reply:
There will never be a case of having fallback hosts with custom ports. If there are custom ports, as per spec, all fallbacks will be empty.